pkg/providers/aws/provider_postgres.go

```diff
@@ -211,8 +211,46 @@ func (p *PostgresProvider) CreatePostgres(ctx context.Context, pg *v1alpha1.Post
 		logger.Infof("created security group %s", aws.StringValue(securityGroup.StandaloneSecurityGroup.GroupName))
 	}
 
+	session := rds.New(sess)
 	// create the aws RDS instance
-	return p.createRDSInstance(ctx, pg, rds.New(sess), ec2.New(sess), rdsCfg, isEnabled)
+	postgres, status, err := p.createRDSInstance(ctx, pg, session, ec2.New(sess), rdsCfg, isEnabled)
+	if err != nil {
+		errMsg := "failed to create rds instance"
+		return nil, status, errorUtil.Wrap(err, errMsg)
+	}
+
+	// Retrieve service maintenance updates, create and export Prometheus metrics
```

Comment:
Ideally this block can go into it's own function or at the bottom of the createRDSInstance function if it works there.